code/modules/mob/living/simple_animal/eldritch_demons.dm

```diff
@@ -0,0 +1,276 @@
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		if(HAS_TRAIT(C, TRAIT_NODISMEMBER))
+			return
+		var/list/parts = list()
+		for(var/X in C.bodyparts)
+			var/obj/item/bodypart/BP = X
+			if(BP.body_part != HEAD && BP.body_part != CHEST && BP.body_part != LEG_LEFT && BP.body_part != LEG_RIGHT)
+				if(BP.dismemberable)
+					parts += BP
+		if(prob(10))
+			var/obj/item/bodypart/BP = pick(parts)
+			if(BP)
+				BP.dismember()
+
+	. = ..()
+
+/mob/living/simple_animal/hostile/eldritch/armsy/prime
+	name = "Lord of the Night"
+	real_name = "Master of Decay"
+	maxHealth = 400
+	health = 400
+	melee_damage_lower = 20
+	melee_damage_upper = 25
+
+/mob/living/simple_animal/hostile/eldritch/armsy/prime/New(spawn_more, len)
+	var/matrix/M = matrix()
+	M.Scale(1.4,1.4)
+	src.transform = M
+	. = ..()
+
+/mob/living/simple_animal/hostile/eldritch/rust_spirit
+	name = "Rust Walker"
+	real_name = "Rusty"
+	desc = "Incomprehensible abomination actively seeping life out of it's surrounding."
+	icon_state = "rust_walker"
+	status_flags = CANPUSH
+	icon_living = "rust_walker"
+	maxHealth = 50
+	health = 50
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	sight = SEE_TURFS
+
+/mob/living/simple_animal/hostile/eldritch/rust_spirit/Initialize()
+	. = ..()
+	AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/rust_conversion/small)
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/projectile/dumbfire/rust_wave/short)
+
+/mob/living/simple_animal/hostile/eldritch/rust_spirit/Life()
+	. = ..()
+	var/turf/T = get_turf(src)
+	if(istype(T,/turf/open/floor/plating/rust))
+		adjustBruteLoss(-3)
+		adjustFireLoss(-3)
+		adjustToxLoss(-3)
+		adjustOxyLoss(-1)
+
+/mob/living/simple_animal/hostile/eldritch/ash_spirit
+	name = "Ash Man"
+	real_name = "Ashy"
+	desc = "Incomprehensible abomination actively seeping life out of it's surrounding."
+	icon_state = "ash_walker"
+	status_flags = CANPUSH
+	icon_living = "ash_walker"
+	maxHealth = 75
+	health = 75
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	sight = SEE_TURFS
+
+/mob/living/simple_animal/hostile/eldritch/ash_spirit/Initialize()
+	. = ..()
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash)
+	AddSpell(new /obj/effect/proc_holder/spell/pointed/ash_cleave/long)
+	AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/fire_cascade)
+
+/mob/living/simple_animal/hostile/eldritch/stalker
+	name = "Flesh Stalker"
+	real_name = "Flesh Stalker"
+	desc = "Abomination made from severed limbs."
+	icon_state = "stalker"
+	status_flags = CANPUSH
+	icon_living = "stalker"
+	maxHealth = 150
+	health = 150
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	sight = SEE_MOBS
+
+/mob/living/simple_animal/hostile/eldritch/stalker/Initialize()
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash)
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/shapeshift/eldritch)
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/emplosion/eldritch)
+	. = ..()
```

Comment:
```suggestion
/mob/living/simple_animal/hostile/eldritch/stalker/Initialize()
	. = ..()
	AddSpell(new /obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash)
	AddSpell(new /obj/effect/proc_holder/spell/targeted/shapeshift/eldritch)
	AddSpell(new /obj/effect/proc_holder/spell/targeted/emplosion/eldritch)
```
Favor calling the parent proc at the start of initialization procs, and at the end of finish-wrapping-up procs.